### PR TITLE
Simplify the quickstart index template

### PR DESCRIPTION
**Pull request by timhoffm**
- Shorten title to "{{ project }} documentation" rather than "Welcome to {{ project }}'s documentation"
- Remove indices and tables
  - Most (in particular smaller) pages do not have a genindex.
  - IMHO modindex is not really useful. If you have API docs, you'll have your own structure there.
  - A link to the (empty) search page is not needed. You can start the search from the sidebar, which will bring you to the search page already populated with the relevant content.
- Add a sentence as example for regular text content.

### CODE REVIEW

File: sphinx/templates/quickstart/root_doc.rst_t [hunk 1]
```diff
@@ -3,19 +3,16 @@
    You can adapt this file completely to your liking, but it should at least
    contain the root `toctree` directive.
 
-Welcome to {{ project }}'s documentation!
-==========={{ project_underline }}=================
+{{ project }} documentation
+{{ project_underline }}==============
+
+Add your content using ``reStructuredText`` syntax. See the
```
Comment (picnixz) [inline sphinx/templates/quickstart/root_doc.rst_t]: I would perhaps put this one in RST comments.

File: sphinx/templates/quickstart/root_doc.rst_t [hunk 2]
```diff
@@ -3,19 +3,16 @@
...
+`reStructuredText <https://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html>`_
+documentation for details.
+
 
 .. toctree::
    :maxdepth: {{ mastertocmaxdepth }}
    :caption: Contents:
 
 {{ mastertoctree }}
-
```
Comment (picnixz) [inline sphinx/templates/quickstart/root_doc.rst_t]: I personally like the fact that we have indices and tables so I would not like to have it removed. In addition, it might be a bit hard for new users to remember those ones.

So instead, I'd suggest putting them in an RST comment *or* keep it altogether. Or maybe, only keep the search part and the others as comments.

File: sphinx/templates/quickstart/root_doc.rst_t (hunk 1 above)
Comment (timhoffm) [reply sphinx/templates/quickstart/root_doc.rst_t]: **Context:**

My plan is to modify "getting started" page to get people quickly to a rendered documentation (PR pending):

image

IMHO seeing the rendered result early is helpful because You have a complete high-level picture (instead of first learning how to add content without knowing where this all is going). Additionally, I think the rendered result is a big motivator to continue.

**This suggestion:**

Adding this text serves two purposes:

- Contained information: Give people a hint how to continue
- Format: By embedding a paragraph of reST with simple markup, people see how the the used markup transfers to the rendered docs.
  In contrast, IMHO comment markup is less intuitive to understand. First, because it does not have any rendered result. Second, because the comment format in reST is a bit particular and unlike comment formats in other languages. We'd have (shortened):
  ```
  .. Add your content using ``reStructuredText`` syntax.

  .. toctree::
     :maxdepth: 2
  ```
  Both, the comment and the toctree directive start with `.. `. For a beginner, it's not trivial to understand the difference.


  
  
   (Comments and directives look alike because they both start with `.. `. Realizing the difference whether that a comment is

File: sphinx/templates/quickstart/root_doc.rst_t (hunk 2 above)
Comment (timhoffm) [reply sphinx/templates/quickstart/root_doc.rst_t]: What benefit do they have for the user? IMHO

- ``:ref:`genindex` `` is more suited for printed documentation. For everything that's searchable (HTML, PDF) a search is usually more convenient and faster than scanning through an alphabetic index.
- ``:ref:`modindex` `` links to a missing page by default --> not a good user experience. And even for cases where you have added API documentation, you typically have only one or very few modules. There's not much to learn from this (e.g. https://requests.readthedocs.io/en/latest/py-modindex/).
- A link to the (empty) search page is not needed. You can start the search from the sidebar, which is more intuitive as every web page search works like this.

While it's great that this functionality exists, I suggest the applications are rather niche and not something we should clutter the default minimal template with.

Basically all documentation pages I know, including our own [sphinx](https://www.sphinx-doc.org/en/master/) and [alabaster](https://alabaster.readthedocs.io/en/latest/) pages do not contain "Indices and Tables".

Comment (AA-Turner) [reply sphinx/templates/quickstart/root_doc.rst_t]: I think Bénédikt has a point in that many features of Sphinx can be hard to find, and the indicies and search from Sphinx are key features. This is a trade-off, though, and it may be that it is worth removing the whole section from the template.

A

**Comment by AA-Turner** [comment]
Let's trial this for now -- we can restore the indicies and tables if user confusion increases.

A

**Comment by AA-Turner** [comment]
Thanks @timhoffm!
